tests/mir-opt/pre-codegen/loops.rs

```diff
@@ -0,0 +1,54 @@
+// compile-flags: -O -Zmir-opt-level=2 -g
+// ignore-debug
+
+#![crate_type = "lib"]
+
+pub fn int_range(start: usize, end: usize) {
+    for i in start..end {
+        opaque(i)
+    }
+}
+
+pub fn vec_range(mut v: Vec<impl Sized>) {
+    for i in 0..v.len() {
+        let x = &mut v[i];
+        opaque((i, x))
+    }
+    for i in 0..v.len() {
+        let x = &v[i];
+        opaque((i, x))
+    }
+}
+
+pub fn vec_iter(mut v: Vec<impl Sized>) {
+    for x in v.iter_mut() {
+        opaque(x)
+    }
+    for x in v.iter() {
+        opaque(x)
+    }
+}
+
+pub fn vec_iter_enumerate(mut v: Vec<impl Sized>) {
```

Comment:
Similarly here, how much of the intent is `Vec` relevant here?  The canonical way to write this would be as `&mut [impl Sized]`, so the `Vec` seems like a distraction.